### DISCUSSION TITLE
ERROR:    Error getting request body

### DISCUSSION THREAD

**Discussion by fishs-x** [fastapi/fastapi#7487] [Questions]
### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI, Body
from pydantic import BaseModel
from fastapi.requests import Request

app = FastAPI()


class Name(BaseModel):
    name: str


@app.middleware("http")
async def add_process_time_header(request: Request, call_next):
    body = await request.body()
    print(f"Body: {body}")
    response = await call_next(request)
    return response


@app.post('/')
def root(name: Name):
    return {"hello": name.name}

```

### Description

```bash

curl -X POST "http://127.0.0.1:8000/" -H "accept: application/json" -H "Content-Type: application/json" -d '{"name": 111}'

* ERROR:    Error getting request body:
```
* Program blocked

### Environment

* OS: [e.g. Linux / Windows / macOS]: macOS
* FastAPI Version [e.g. 0.3.0]:  0.54.1
* Python version: Python 3.7.4

**Comment by igotinfected** [comment]
If I understood this correctly the last time this came up, `await`ing the request body in the middleware **consumes** it, which means it becomes unusable after. You can try the solution proposed [here](https://github.com/tiangolo/fastapi/issues/394#issuecomment-513051977) as a workaround (no time to test it myself currently).

For an even better approach, you can look into https://github.com/tiangolo/fastapi/pull/589, or into the [documentation](https://fastapi.tiangolo.com/advanced/custom-request-and-route/) version of that pull request.

**Comment by fishs-x** [comment]
@IgnatovFedor Thanks

**Comment by tiangolo** [comment]
Thanks for the help here @igotinfected ! :clap: :bow: 

Thanks for reporting back and closing the issue @fishs-x  :+1:

**Comment by YuriiMotov** [comment]
This was solved in [Starlette 0.28](https://www.starlette.io/release-notes/#0280-june-7-2023)
